### DISCUSSION TITLE
Error loading remote context

### DISCUSSION THREAD

**Discussion by spartahawk** [linked-data-dotnet/json-ld.net#150] [Q&A]
I'm trying to make use of [json-ld.net](http://json-ld.net/) expansion and contraction given the json-ld formatting I'm seeing from major retailers. Are you aware of a discrepancy that follows?

In the repo readme, the example for expansion in the installation uses this example for the json-ld to be expanded:

```jsonld
{"@context":{ "test":"http://www.example.org/"},"test:hello":"world"}
```

However the major retailers I've looked at have it in something like the following format:

```jsonld
{"@context":"https://schema.org/", "@type":"Product", "image":"https://exampleimage.jpeg/",
   "name":"fake product name", "sku":"2345624623", ...}
```

So rather than have context be an object containing name/value pairs with what looks like individualized contexts for each thing we could expand, these retailers just list a single context url that I assume is supposed to be applied to everything else in the json-ld.

The `Expand()` method doesn't properly process inputs that don't have context as an object. It gives the error: error loading remote context 

So my question is whether I'm needing to provide something more for Expand() to properly assign context to each name/value pair in the json-ld contents based on the single context url these retailers provide, or if I'll have to tweak the incoming json-ld format in order to feed it to the expand algorithm in a way it can process properly.

Please let me know if I'm misunderstanding. I'd be happy to hop on a quick call via Zoom or whatever if that's fastest for you. Thanks so much.

The way this library is expecting the context to be formatted doesn't seem to match google's structured data examples for products, nor any of the retailers I've looked at. So it's a bit confusing. https://developers.google.com/search/docs/advanced/structured-data/product#single-product-page

**Comment by sblom** [comment]
That appears to be the result of JSON-LD.NET not yet supporting JSON-LD 1.1. If you try your example in the [JSON-LD 1.0 playground](https://json-ld.org/playground/1.0/#startTab=tab-expanded&json-ld=%7B%22%40context%22%3A%22https%3A%2F%2Fschema.org%2F%22%2C%22%40type%22%3A%22Product%22%2C%22image%22%3A%22https%3A%2F%2Fexampleimage.jpeg%2F%22%2C%22name%22%3A%22fake%20product%20name%22%2C%22sku%22%3A%222345624623%22%7D), you get a very similar failure.

**Reply by spartahawk** [reply]
Thanks @sblom . Any suggested workaround? 1.1 is everywhere now.

**Reply by sblom** [reply]
Unfortunately I haven't even looked at 1.1 (I've been mostly away for a while). If the only problem is the @context syntax, I imagine a trivial preprocessing step could get you to a 1.0-style context description.

The ubiquity of 1.1 is a good point. I don't know of any current efforts to modernize this library, but I would heavily assist anyone who was motivated.
